### PR TITLE
[opentitantool] Implement image digest command

### PR DESCRIPTION
**Pull request by jon-flatley**
Implementation for the `opentitantool image digest` command. Given an input image this command will produce a digest for the signed region of that image.

### CODE REVIEW

File: sw/host/opentitantool/src/command/image.rs [hunk 1]
```diff
@@ -155,7 +155,8 @@ impl CommandDispatch for DigestCommand {
         _context: &dyn Any,
         _transport: &TransportWrapper,
     ) -> Result<Option<Box<dyn Serialize>>> {
-        Ok(None)
+        let image = image::Image::read_from_file(&self.image)?;
+        Ok(Some(Box::new(format!("{}", image.compute_digest()))))
```
Comment (cfrantz) [inline sw/host/opentitantool/src/command/image.rs:159]: Lets define a struct above like:
```
#[derive(serde::Serialize)]
struct DigestResponse {
    pub digest: String,
}
```

And then return that struct here:
```
Ok(some(Box::new(DigestResponse {
    digest: format!(...),
})))
```

Lets also add an `bin: Option<PathBuf>` to `DigestCommand` and use that to write the digest out as a binary file:
```
if let Some(bin) = &self.bin {
    write_to_file(bin, image.compute_digest())?;
}
```

I think we'll want the binary representation available for the HSM based signing infrastructure.
WDYT?

File: sw/host/opentitanlib/src/image/image.rs [hunk 2]
```diff
@@ -4,8 +4,11 @@
 
 use crate::image::manifest::Manifest;
 use crate::image::manifest_def::ManifestDef;
+use crate::util::bigint;
 use crate::util::parse_int::ParseInt;
 use anyhow::{ensure, Result};
+use memoffset::offset_of;
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/image/image.rs:10]: style nit: extern and std crates first, then local `crate::`s.

**Review by cfrantz — COMMENTED** [review]
Looks good, just a couple of comments.

File: sw/host/opentitanlib/src/image/image.rs [hunk 3]
```diff
@@ -97,6 +103,13 @@ impl Image {
         *manifest = manifest_def.try_into()?;
         Ok(())
     }
+
+    /// Compute the SHA256 digest for the signed portion of the `Image`.
+    pub fn compute_digest(&self) -> Sha256Hash {
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/image/image.rs:108]: TODO: Lets add a test for this.  Ok to do in a follow-up PR.

**Review by cfrantz — APPROVED** [review]

**Comment by jon-flatley** [comment]
@cfrantz I think this also suffers from the issue I pointed out earlier with `bigint` if the MSB of the digest is 0. I'm going to drop the usage of `bigint` in this case.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**